pkg/storage/loaders/data_loader_middleware.go

```diff
@@ -26,9 +26,15 @@ func NewDataLoaderMiddleware(loaders *DataLoaders) func(http.Handler) http.Handl
 	}
 }
 
-// Loaders returns the dataLoaders for a given context
-func Loaders(ctx context.Context) *DataLoaders {
-	return ctx.Value(loadersKey).(*DataLoaders)
+// Loaders returns the dataLoaders for a given context, along with a bool to say if it returned. It will return true if the loader is present, and false if not
+func Loaders(ctx context.Context) (*DataLoaders, bool) {
+	loaders := ctx.Value(loadersKey)
+	if loaders == nil {
+		return nil, false
+	}
```

Comment:
I think every call I found to this function ends up returning `ErrNoLoaderOnContext` -- maybe we should just return `(*DataLoaders, err)` from this function instead of a bool, and then we can just return that error in all those places instead of them each referencing `ErrNoLoaderOnContext`?